Upgrade EUI to v95.12.0

cee-chen opened the pull request:
`v95.11.0`⏩`v95.12.0-backport.0`

> [!note]
> A few fun highlights from this release:
> - Content within `EuiDataGrid`, `EuiBasicTable`, and `EuiInMemoryTable`, when manually selected/highlighted by users, should now more cleanly copy and paste into various spreadsheet apps
> - `EuiProvider` will now detect the user's system-level dark or light mode and default to that if no `colorMode` prop is passed manually

_[Questions? Please see our Kibana upgrade FAQ.](https://github.com/elastic/eui/blob/main/wiki/eui-team-processes/upgrading-kibana.md#faq-for-kibana-teams)_

---

## [`v95.12.0`](https://github.com/elastic/eui/releases/v95.12.0)

- Enhanced `EuiDataGrid` and `EuiBasic/InMemoryTable` to clean content newlines/tabs when users copy and paste from their tabular data ([#8019](https://github.com/elastic/eui/pull/8019))
- Updated `EuiResizableButton` with a new `accountForScrollbars` prop ([#8021](https://github.com/elastic/eui/pull/8021))
- Updated `EuiProvider` to inherit from the user's OS/system light/dark mode setting if a `colorMode` prop has not been passed ([#8026](https://github.com/elastic/eui/pull/8026))

**Bug fixes**

- Fixed `EuiDatePicker`'s `onClear` button to not appear when the input is `disabled` ([#8020](https://github.com/elastic/eui/pull/8020))
- Fixed several `EuiDataGrid` row height bugs: ([#8025](https://github.com/elastic/eui/pull/8025))
  - Fixed row heights not recalculating when `rowHeightOptions.lineHeight`, `gridStyles.fontSize`, or `gridStyles.cellPadding` changed
  - Fixed incorrect height calculations for `rowHeightOptions.rowHeights` with `lineCount`s
  - Fixed control column content to align better with multi-line row heights, as well as custom line-heights
## [`v95.12.0-backport.0`](https://github.com/elastic/eui/releases/v95.12.0-backport.0)

**This is a backport release only intended for use by Kibana.**

**Bug fixes**

- Fixed `EuiProvider`'s system color mode detection causing errors during server-side rendering ([#8040](https://github.com/elastic/eui/pull/8040))
- Fixed an `EuiDataGrid` rendering bug that was causing bouncing scrollbar issues ([#8041](https://github.com/elastic/eui/pull/8041))

<!--ONMERGE {"backportTargets":["8.x"]} ONMERGE-->

cee-chen committed:
Upgrade EUI to v95.12.0

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!

cee-chen committed:
Backport with patch fixes

cee-chen committed:
Fix Jest `matchMedia` undefined failures

- `jest.fn()` is legitimately not available (or possibly getting overwritten) and causing undefined failures down the road

- just replace `jest.fn()`s with static fns / noops since no one is mocking or listening to them in any case

cee-chen committed:
[FTR] Fix failing text assertions due to hidden newline copy markers

+ harden header control columns to exclude SR text as well

cee-chen committed:
[Cypress] Update queries for text assertions to exclude new hidden copy characters

elasticmachine commented:
Pinging @elastic/eui-team (EUI)

wayneseymour reviewed (approved):
Code Review only, LGTM

peteharverson reviewed (approved):
ML changes LGTM

jbudz commented on packages/kbn-test/src/jest/setup/polyfills.jsdom.js:
```diff
@@ -53,22 +53,16 @@ if (!Object.hasOwn(global, 'Worker')) {
       writable: true,
-      // eslint-disable-next-line no-undef
-      value: jest.fn().mockImplementation((query) => ({
+      value: (query) => ({
```
cc @Dosant, does this work for you?

Dosant replied:
lgtm!

cee-chen replied:
Thank you for checking, both of you!

jughosta commented on test/functional/apps/discover/esql/_esql_view.ts:
```diff
@@ -199,7 +199,7 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
         expect(await cell.getVisibleText()).to.be(' - ');
         expect(await dataGrid.getHeaders()).to.eql([
-          "Select columnPress the Enter key to interact with this cell's contents.", // contains screen reader help text
+          'Select column',
```
Hi @cee-chen,

Sorry, I don't have a chance to review the PR fully this week. From a quick look, it seems like the row controls can get misaligned depending on the display settings:

<img width="1255" alt="Screenshot 2024-09-26 at 17 32 55" src="https://github.com/user-attachments/assets/d250d3d6-8508-4fa5-91eb-c4abcbe9037b">

cee-chen replied:
Will take a look at this, thanks for flagging it Julia!

cee-chen replied:
Fixed now with the latest commits:

<img width="412" alt="" src="https://github.com/user-attachments/assets/cd93b9df-bc6f-499a-99d1-ded2dfa4cc41">

<img width="305" alt="" src="https://github.com/user-attachments/assets/e70ea79f-ddb1-48d3-9397-8c943150ce26">

The EUI changes let me clean up a bunch of extra wrappers while also making the checkboxes/buttons bounce less when switching between single and custom! 🥳

jughosta replied:
Great, thanks! 😍

cee-chen committed:
Merge branch 'main' into eui/v95.12.0

cee-chen committed:
[Discover DataTable] Fix vertical centering of selection checkboxes

- flex groups and extra padding are no longer necessary with the new EuiDataGrid row height fixes

cee-chen committed:
[Discover DataTable] Fix vertical centering of row control buttons

- remove `<DataTableRowControl />` altogether - the size doesn't appear to do anything and the extra `<span>` wrapper interferes with our ability to vertically align w/ line-height

- tweak negative margin-tops by minute amounts, as that's all that's needed at this point

cee-chen committed:
[Cypress] Update queries for text assertions to exclude new hidden copy characters

cee-chen committed:
Update snapshot

cee-chen committed:
Merge remote-tracking branch 'upstream/main' into eui/v95.12.0

paul-tavares reviewed (approved):
Changes look good - team: elastic/security-defend-workflows 

🙏

MichaelMarcialis reviewed (approved):
Code review only. LGTM.

cee-chen committed:
Merge branch 'main' into eui/v95.12.0

jaredburgettelastic reviewed (approved):
Entity Analytics changes LGTM!

kibanamachine commented:
Starting backport for target branches: 8.x

https://github.com/elastic/kibana/actions/runs/11115699066

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/194500">](https://github.com/elastic/kibana/pull/194500)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.4.3 -->
